packages/compass-explain-plan/src/modules/explain.js

```diff
@@ -389,6 +396,8 @@ export const fetchExplainPlan = (query) => {
       explain.resultId = resultId();
       explain.error = error;
       dispatch(explainPlanFetched(explain));
```

Comment:
In other tabs, we hide the content when there's an error. we can have similar behaviour here as well?

Reply:
So I looked at the document tab and schema analysis tab as starting points:
- On error both of them display the error banner, content is gone (or not there already)
- On cancel of operation, document tab displays an error banner with content "Operation aborted" but the schema analysis tab simply switches back to zero content screen which I guess is the welcome page of the feature itself. I found the latter more clear and was also something that I guess Rhys was trying to mention in the comment [here](https://jira.mongodb.org/browse/COMPASS-4620?focusedCommentId=4354087&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4354087).

To move forward I have adapted the behaviour that is there in schema analysis tab - switching back to zero content. Let me know how you feel about that?

Reply:
FWIW resetting to the initial state becomes confusing when you already have results in the view before re-running, because if it's not an error that we are showing I would expect it to go back to the previous state, not the initial empty one. We can add this as a discussion point for the design sync, would be indeed nice to decide on one approach across views.

Reply:
Design team agreed on having the behaviour same as what we have at the moment in aggregation tab, also documented [here](https://jira.mongodb.org/browse/EXPO-2181?focusedCommentId=5160056&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-5160056). Required changes are applied 👍